multiqc/modules/biscuit/biscuit.py

```diff
@@ -64,214 +66,210 @@ def __init__(self):
-            mapq[s[0]] = s[1]  # mapq[MAPQ] = number of reads
-
-        data = {
-            "frc_align": 0,
-            "opt_align": 0,
-            "sub_align": 0,
-            "not_align": 0,
-            "mapqs": dict(zip(range(61), [0 for _ in range(61)])),
-        }
-        if len(mapq) > 0:
-            total = sum([int(cnt) for _, cnt in mapq.items() if _ != "unmapped"])
-            for mq, cnt in mapq.items():
-                if mq == "unmapped":
-                    data["not_align"] += int(cnt)
-                else:
-                    data["mapqs"][int(mq)] = 100.0 * float(cnt) / total
-                    if int(mq) >= 40:
-                        data["opt_align"] += int(cnt)
-                    else:
-                        data["sub_align"] += int(cnt)
-
-            data["frc_align"] = (
-                100
-                * (data["opt_align"] + data["sub_align"])
-                / (data["opt_align"] + data["sub_align"] + data["not_align"])
-            )
-
-        return data
+        self.general_stats_addcols(pd, header)
 
-    ########################################
-    ####    Alignment Quality Report    ####
-    ########################################
+    ################################################################################
+    ##                             Plotting Functions                             ##
+    ################################################################################
     def chart_align_mapq(self):
-        """
-        Chart _mapq_table.txt
-        Inputs:
-            No inputs
-        Returns:
-            No returns, generates Mapping Overview and Mapping Quality
-            Distribution charts
-        """
-
-        #
-        # Mapping Overview bar chart
-        #
-        pd = {}
-
-        # Calculate alignment counts
-        for s_name, dd in self.mdata["align_mapq"].items():
-            if len(dd) > 0:
-                pd[s_name] = {"opt_align": dd["opt_align"], "sub_align": dd["sub_align"], "not_align": dd["not_align"]}
-
-        pheader = {
+        """Mapping overview plots"""
+        # Extract data
+        pd1 = {}
+        pd2 = {}
+        for s_name, d in self.biscuit_data["align_mapq"].items():
+            if len(d) > 0:
+                pd1[s_name] = {
+                    "opt_align": d["opt_align"],
+                    "sub_align": d["sub_align"],
+                    "not_align": d["not_align"],
+                }
+                pd2[s_name] = d["mapqs"]
+
+        # Mapping Overview bar graph
+        pheader1 = {
             "opt_align": {"color": "#1f78b4", "name": "Optimally Aligned Reads"},
             "sub_align": {"color": "#a6cee3", "name": "Suboptimally Aligned Reads"},
             "not_align": {"color": "#b2df8a", "name": "Unaligned Reads"},
         }
 
-        pconfig = {
-            "id": "biscuit-mapping-overview-plot",
+        pconfig1 = {
+            "id": "biscuit_mapping_overview",
```

Comment:
You have duplicate IDs apart from `_` / `-` separators here, which is a bit of a bad code smell. I think that the previous `-plot` suffix is better:

```suggestion
            "id": "biscuit-mapping-overview-plot",
```